pkg/services/accesscontrol/accesscontrol.go

```diff
@@ -120,3 +124,39 @@ func ValidateScope(scope string) bool {
 	}
 	return !strings.ContainsAny(prefix, "*?")
 }
+
+func addActionToMetadata(allMetadata map[string]Metadata, action, id string) map[string]Metadata {
+	metadata, initialized := allMetadata[id]
+	if !initialized {
+		metadata = Metadata{action: true}
+	} else {
+		metadata[action] = true
+	}
+	allMetadata[id] = metadata
+	return allMetadata
+}
+
+// GetResourcesMetadata returns a map of accesscontrol metadata, listing for each resource, users available actions
+func GetResourcesMetadata(ctx context.Context, permissions []*Permission, resource string, resourceIDs map[string]bool) (map[string]Metadata, error) {
+	allScope := GetResourceAllScope(resource)
+	allIDScope := GetResourceAllIDScope(resource)
+
+	// Loop through permissions once
+	result := map[string]Metadata{}
+	for _, p := range permissions {
+		if p.Scope == "*" || p.Scope == allScope || p.Scope == allIDScope {
+			// Add global action to all resources
+			for id := range resourceIDs {
+				result = addActionToMetadata(result, p.Action, id)
+			}
+		} else {
+			parts := strings.Split(p.Scope, ":")
+			if len(parts) == 3 && parts[0] == resource && parts[1] == "id" && resourceIDs[parts[2]] {
```

Comment:
This is not going to work for resources with more than one field, ex: `global:users`. :thinking:

Reply:
Ok, thought about this a little bit more. I think it's ok to merge as is and figure out how to change this when the need actually comes. Wdot?

Reply:
All scopes has to either be suffixed with '*' or an id. So i think you could always compare take the last part of a scope as id. Then `global:users:id:10` would not be a problem right?

Reply:
Yes, just wanted to make sure the prefix matched as well :+1: